crates/red_knot_module_resolver/build.rs

```diff
@@ -28,25 +29,25 @@ fn zip_dir(directory_path: &str, writer: File) -> ZipResult<File> {
 
     for entry in walkdir::WalkDir::new(directory_path) {
         let dir_entry = entry.unwrap();
-        let relative_path = dir_entry.path();
-        let name = relative_path
+        let absolute_path = dir_entry.path();
+        let normalized_relative_path = absolute_path
             .strip_prefix(Path::new(directory_path))
             .unwrap()
-            .to_str()
+            .to_slash()
             .expect("Unexpected non-utf8 typeshed path!");
 
         // Write file or directory explicitly
         // Some unzip tools unzip files with directory paths correctly, some do not!
-        if relative_path.is_file() {
-            println!("adding file {relative_path:?} as {name:?} ...");
-            zip.start_file(name, options)?;
-            let mut f = File::open(relative_path)?;
+        if absolute_path.is_file() {
+            println!("adding file {absolute_path:?} as {normalized_relative_path:?} ...");
+            zip.start_file(normalized_relative_path, options)?;
```

Comment:
There's a `start_file_from_path` function that takes care of the normalization

Reply:
On v0.6.6, the version we have pinned in `Cargo.toml`, there is a deprecation warning attached to this method in the docs: https://docs.rs/zip/0.6.6/zip/write/struct.ZipWriter.html#method.start_file_from_path.

The deprecation warning isn't there on the latest version, so I suppose they must have fixed the issues and undeprecated the method. But we're keeping `zip` pinned to the old version due to https://github.com/astral-sh/uv/issues/3642

Reply:
This is what the change looks like if we want to keep the `zip` crate pinned to v0.6.6 and have Clippy pass:

```diff
--- a/crates/red_knot_module_resolver/build.rs
+++ b/crates/red_knot_module_resolver/build.rs
@@ -8,7 +8,6 @@
 use std::fs::File;
 use std::path::Path;
 
-use path_slash::PathExt;
 use zip::result::ZipResult;
 use zip::write::{FileOptions, ZipWriter};
 use zip::CompressionMethod;
@@ -32,22 +31,22 @@ fn zip_dir(directory_path: &str, writer: File) -> ZipResult<File> {
         let absolute_path = dir_entry.path();
         let normalized_relative_path = absolute_path
             .strip_prefix(Path::new(directory_path))
-            .unwrap()
-            .to_slash()
-            .expect("Unexpected non-utf8 typeshed path!");
+            .unwrap();
 
         // Write file or directory explicitly
         // Some unzip tools unzip files with directory paths correctly, some do not!
         if absolute_path.is_file() {
             println!("adding file {absolute_path:?} as {normalized_relative_path:?} ...");
-            zip.start_file(normalized_relative_path, options)?;
+            #[allow(deprecated)]
+            zip.start_file_from_path(normalized_relative_path, options)?;
             let mut f = File::open(absolute_path)?;
             std::io::copy(&mut f, &mut zip).unwrap();
-        } else if !normalized_relative_path.is_empty() {
+        } else if normalized_relative_path == Path::new("") {
             // Only if not root! Avoids path spec / warning
             // and mapname conversion failed error on unzip
             println!("adding dir {absolute_path:?} as {normalized_relative_path:?} ...");
-            zip.add_directory(normalized_relative_path, options)?;
+            #[allow(deprecated)]
+            zip.add_directory_from_path(normalized_relative_path, options)?;
         }
     }
```